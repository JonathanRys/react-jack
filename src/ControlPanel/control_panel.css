.ControlPanel_main {
  background: #eee;
  border: 1px solid black;
  border-radius: 10px;
  width: 100%;
  min-width: 240px;
  margin-top: 20px;
  padding: 25px;
}

.ControlPanel_main_panel-header {
  font-size: 1.5em;
  font-weight: bold;
}
