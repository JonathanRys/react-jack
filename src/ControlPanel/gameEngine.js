export default function gameEngine(nextProps) {
  const index = nextProps.player.handIndex;
  const dealersTurn =
    nextProps.player.playerStands[index] ||
    nextProps.player.busted[index] ||
    nextProps.player.hasBlackjack[index];

  const dealerScore = nextProps.dealer.score;
  const playerScore = nextProps.player.score[index];

  // Don't use the game engine if the game is over
  if (!nextProps.turn.isPlaying) return null;

  // If there is a drawn card, take it, clear it and await new props
  if (nextProps.deck.drawnCard) {
    if (nextProps.turn.playersTurn) {
      nextProps.giveCard(nextProps.deck.drawnCard);
    } else {
      nextProps.giveDealerCard(nextProps.deck.drawnCard);
    }
    return null;
  }

  // Since we're playing, see if the players have all their cards
  if (
    nextProps.dealer.hand.length < 2 ||
    nextProps.player.hands[0].length < 2
  ) {
    // Deal the cards
    nextProps.keepDealing();
    return null;
  }

  if (nextProps.turn.playersTurn) {
    // It's the player's turn, but check if that should change
    if (dealersTurn) {
      nextProps.dealerTurn();
    }
    return null;
  } else if (dealersTurn) {
    // It's still the dealer's turn - hit until 17
    if (nextProps.dealer.score < 17) {
      // Don't bother drawing if the player is busted or has blackjack
      if (nextProps.player.busted[index]) {
        nextProps.stop();
        return "Bust";
      } else if (nextProps.player.hasBlackjack[index]) {
        nextProps.stop();
        nextProps.winBet(2.5);
        return "Blackjack";
      }
      // Draw a card
      nextProps.hitOnClick();
      return null;
    } else {
      // Stop playing and...
      nextProps.stop();
      // Score the player's hand
      if (playerScore > 21) {
        return "Bust";
      } else if (playerScore === dealerScore) {
        nextProps.winBet();
        return "Push";
      } else if (nextProps.player.hasBlackjack[index]) {
        nextProps.winBet(2.5);
        return "Blackjack";
      } else if (dealerScore > 21 || playerScore > dealerScore) {
        nextProps.winBet(2);
        return "Won";
      } else {
        return "Lost";
      }
    }
  } else {
    // The turn has ended, awaiting user input
    nextProps.nextPlayer();
    return null;
  }
}
