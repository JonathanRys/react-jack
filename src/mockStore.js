export default {
  dealer: {
    name: "Dealer",
    avatar: "./images/dealer.png",
    hand: ["HA", "C5"],
    score: 0,
    busted: false,
    hasBlackjack: false
  },
  player: {
    name: "Player1",
    avatar: "./images/default.png",
    playerIndex: 1,

    balance: 500,
    currentBet: 5,

    hands: [["SA", "CK"], ["D2", "HJ"]],
    status: [null],
    score: [0],
    busted: [false],
    hasBlackjack: [false],
    hasInsurance: [false],
    splitHand: false,
    playerStands: [false]
  },
  turn: {
    isPlaying: false,
    numPlayers: 2,
    playersTurn: 1,
    error: false,
    errMsg: ""
  },
  deck: {
    drawnCard: null,
    deck: []
  },
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn()
};
