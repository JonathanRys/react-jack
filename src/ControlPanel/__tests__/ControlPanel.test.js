import React from "react";
import { shallow } from "enzyme";

import ControlPanel from "../ControlPanel";

import mockStore from "../../mockStore";

const mockProps = {
  ...mockStore,
  setStatus: jest.fn(),
  hitOnClick: jest.fn(),
  standOnClick: jest.fn(),
  winBet: jest.fn(),
  loseBet: jest.fn(),
  splitOnClick: jest.fn(),
  doubleDownOnClick: jest.fn(),
  buyInsuranceOnClick: jest.fn(),
  dealOnClick: jest.fn(),
  reset: jest.fn()
};

// test every permutation of state that applies to the game to make sure it operates as expected

describe("Test PlayField component", () => {
  it("matches the snapshot", () => {
    const testProps = { ...mockProps };
    const component = shallow(<ControlPanel {...testProps} />);
    expect(component).toMatchSnapshot();
  });

  it("test snapshot with isPlaying equal to true", () => {
    const testProps = { ...mockProps };
    testProps.turn.isPlaying = true;
    const component = shallow(<ControlPanel {...testProps} />);
    expect(component).toMatchSnapshot();
  });

  it("updates only when the props have changed", () => {
    const testProps = { ...mockProps };
    const component = shallow(<ControlPanel {...testProps} />);
    const shouldComponentUpdate = component.instance().shouldComponentUpdate;

    expect(shouldComponentUpdate.bind({ props: true })(true)).toEqual(false);
    expect(shouldComponentUpdate.bind({ props: true })(false)).toEqual(true);
    expect(shouldComponentUpdate.bind({ props: false })(true)).toEqual(true);
    expect(shouldComponentUpdate.bind({ props: false })(false)).toEqual(false);
  });

  it("dummy test for componentWillUpdate", () => {
    const testProps = { ...mockProps };
    testProps.turn.isPlaying = false;
    const component = shallow(<ControlPanel {...testProps} />);
    const componentWillUpdate = component.instance().componentWillUpdate;

    expect(componentWillUpdate(testProps)).toEqual(undefined);
  });

  it("sets the status if the componentWillUpdate", () => {
    const testProps = { ...mockProps };
    testProps.turn.isPlaying = true;
    const component = shallow(<ControlPanel {...testProps} />);
    const componentWillUpdate = component.instance().componentWillUpdate;

    expect(componentWillUpdate(testProps)).toEqual(undefined);
    expect(testProps.setStatus).toHaveBeenCalledWith(null);
  });

  it("ends the turn with buyInsuranceOnClick", () => {
    const testProps = { ...mockProps };
    testProps.turn.isPlaying = true;
    const component = shallow(<ControlPanel {...testProps} />);
    const buyInsuranceOnClick = component.instance().buyInsuranceOnClick;

    buyInsuranceOnClick();

    expect(testProps.buyInsuranceOnClick).toHaveBeenCalled();
    expect(testProps.standOnClick).toHaveBeenCalled();
  });

  it("switches onClick handlers when the dealer has blackjack", () => {
    const testProps = { ...mockProps };
    testProps.turn.isPlaying = true;
    testProps.dealer.hand = ["SA", "HJ"];
    testProps.dealer.hasBlackjack = true;
    const component = shallow(<ControlPanel {...testProps} />);

    expect(component).toMatchSnapshot();
  });
});
