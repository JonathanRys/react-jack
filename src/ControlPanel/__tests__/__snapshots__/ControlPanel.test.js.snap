// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlayField component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlPanel
    buyInsuranceOnClick={[MockFunction]}
    dealOnClick={[MockFunction]}
    dealer={
      Object {
        "avatar": "./images/dealer.png",
        "busted": false,
        "hand": Array [
          "HA",
          "C5",
        ],
        "hasBlackjack": false,
        "name": "Dealer",
        "score": 0,
      }
    }
    deck={
      Object {
        "deck": Array [],
        "drawnCard": null,
      }
    }
    dispatch={[MockFunction]}
    doubleDownOnClick={[MockFunction]}
    getState={[MockFunction]}
    hitOnClick={[MockFunction]}
    loseBet={[MockFunction]}
    player={
      Object {
        "avatar": "./images/default.png",
        "balance": 500,
        "busted": Array [
          false,
        ],
        "currentBet": 5,
        "hands": Array [
          Array [
            "SA",
            "CK",
          ],
          Array [
            "D2",
            "HJ",
          ],
        ],
        "hasBlackjack": Array [
          false,
        ],
        "hasInsurance": Array [
          false,
        ],
        "name": "Player1",
        "playerIndex": 1,
        "playerStands": Array [
          false,
        ],
        "score": Array [
          0,
        ],
        "splitHand": false,
        "status": Array [
          null,
        ],
      }
    }
    reset={[MockFunction]}
    setStatus={[MockFunction]}
    splitOnClick={[MockFunction]}
    standOnClick={[MockFunction]}
    subscribe={[MockFunction]}
    turn={
      Object {
        "errMsg": "",
        "error": false,
        "isPlaying": false,
        "numPlayers": 2,
        "playersTurn": 1,
      }
    }
    winBet={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Button
          block={true}
          color="secondary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-o-down"
          />
           Hit
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-paper-o"
          />
           Stand
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-scissors-o"
          />
           Split
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="align-justify "
          />
           Double Down
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="dollar"
          />
           Buy Insurance
        </Button>,
        <Button
          block={true}
          color="primary"
          disabled={false}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="handshake-o"
          />
           Deal
        </Button>,
      ],
      "className": "ControlPanel_main",
      "lg": "12",
      "md": "8",
      "sm": "6",
      "tag": "form",
      "xs": "4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-o-down"
            />,
            " Hit",
          ],
          "color": "secondary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-o-down",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Hit",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-paper-o"
            />,
            " Stand",
          ],
          "color": "secondary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-paper-o",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Stand",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-scissors-o"
            />,
            " Split",
          ],
          "color": "secondary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-scissors-o",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Split",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="align-justify "
            />,
            " Double Down",
          ],
          "color": "secondary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "align-justify ",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Double Down",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="dollar"
            />,
            " Buy Insurance",
          ],
          "color": "secondary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "dollar",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Buy Insurance",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="handshake-o"
            />,
            " Deal",
          ],
          "color": "primary",
          "disabled": false,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "handshake-o",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Deal",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            block={true}
            color="secondary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-o-down"
            />
             Hit
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-paper-o"
            />
             Stand
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-scissors-o"
            />
             Split
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="align-justify "
            />
             Double Down
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="dollar"
            />
             Buy Insurance
          </Button>,
          <Button
            block={true}
            color="primary"
            disabled={false}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="handshake-o"
            />
             Deal
          </Button>,
        ],
        "className": "ControlPanel_main",
        "lg": "12",
        "md": "8",
        "sm": "6",
        "tag": "form",
        "xs": "4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-o-down"
              />,
              " Hit",
            ],
            "color": "secondary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-o-down",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Hit",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-paper-o"
              />,
              " Stand",
            ],
            "color": "secondary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-paper-o",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Stand",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-scissors-o"
              />,
              " Split",
            ],
            "color": "secondary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-scissors-o",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Split",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="align-justify "
              />,
              " Double Down",
            ],
            "color": "secondary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "align-justify ",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Double Down",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="dollar"
              />,
              " Buy Insurance",
            ],
            "color": "secondary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "dollar",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Buy Insurance",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="handshake-o"
              />,
              " Deal",
            ],
            "color": "primary",
            "disabled": false,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "handshake-o",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Deal",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test PlayField component test isPlaying 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlPanel
    buyInsuranceOnClick={[MockFunction]}
    dealOnClick={[MockFunction]}
    dealer={
      Object {
        "avatar": "./images/dealer.png",
        "busted": false,
        "hand": Array [
          "HA",
          "C5",
        ],
        "hasBlackjack": false,
        "name": "Dealer",
        "score": 0,
      }
    }
    deck={
      Object {
        "deck": Array [],
        "drawnCard": null,
      }
    }
    dispatch={[MockFunction]}
    doubleDownOnClick={[MockFunction]}
    getState={[MockFunction]}
    hitOnClick={[MockFunction]}
    loseBet={[MockFunction]}
    player={
      Object {
        "avatar": "./images/default.png",
        "balance": 500,
        "busted": Array [
          false,
        ],
        "currentBet": 5,
        "hands": Array [
          Array [
            "SA",
            "CK",
          ],
          Array [
            "D2",
            "HJ",
          ],
        ],
        "hasBlackjack": Array [
          false,
        ],
        "hasInsurance": Array [
          false,
        ],
        "name": "Player1",
        "playerIndex": 1,
        "playerStands": Array [
          false,
        ],
        "score": Array [
          0,
        ],
        "splitHand": false,
        "status": Array [
          null,
        ],
      }
    }
    reset={[MockFunction]}
    setStatus={[MockFunction]}
    splitOnClick={[MockFunction]}
    standOnClick={[MockFunction]}
    subscribe={[MockFunction]}
    turn={
      Object {
        "errMsg": "",
        "error": false,
        "isPlaying": true,
        "numPlayers": 2,
        "playersTurn": 1,
      }
    }
    winBet={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Button
          block={true}
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-o-down"
          />
           Hit
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-paper-o"
          />
           Stand
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-scissors-o"
          />
           Split
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="align-justify "
          />
           Double Down
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="dollar"
          />
           Buy Insurance
        </Button>,
        <Button
          block={true}
          color="primary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="handshake-o"
          />
           Deal
        </Button>,
      ],
      "className": "ControlPanel_main",
      "lg": "12",
      "md": "8",
      "sm": "6",
      "tag": "form",
      "xs": "4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-o-down"
            />,
            " Hit",
          ],
          "color": "secondary",
          "disabled": false,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-o-down",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Hit",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-paper-o"
            />,
            " Stand",
          ],
          "color": "secondary",
          "disabled": false,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-paper-o",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Stand",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-scissors-o"
            />,
            " Split",
          ],
          "color": "secondary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-scissors-o",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Split",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="align-justify "
            />,
            " Double Down",
          ],
          "color": "secondary",
          "disabled": false,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "align-justify ",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Double Down",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="dollar"
            />,
            " Buy Insurance",
          ],
          "color": "secondary",
          "disabled": false,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "dollar",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Buy Insurance",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="handshake-o"
            />,
            " Deal",
          ],
          "color": "primary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "handshake-o",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " Deal",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            block={true}
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-o-down"
            />
             Hit
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-paper-o"
            />
             Stand
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-scissors-o"
            />
             Split
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="align-justify "
            />
             Double Down
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="dollar"
            />
             Buy Insurance
          </Button>,
          <Button
            block={true}
            color="primary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="handshake-o"
            />
             Deal
          </Button>,
        ],
        "className": "ControlPanel_main",
        "lg": "12",
        "md": "8",
        "sm": "6",
        "tag": "form",
        "xs": "4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-o-down"
              />,
              " Hit",
            ],
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-o-down",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Hit",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-paper-o"
              />,
              " Stand",
            ],
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-paper-o",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Stand",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-scissors-o"
              />,
              " Split",
            ],
            "color": "secondary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-scissors-o",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Split",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="align-justify "
              />,
              " Double Down",
            ],
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "align-justify ",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Double Down",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="dollar"
              />,
              " Buy Insurance",
            ],
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "dollar",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Buy Insurance",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="handshake-o"
              />,
              " Deal",
            ],
            "color": "primary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "handshake-o",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " Deal",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
